internal/controllers/extension_controller.go

```diff
@@ -162,35 +164,47 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 	ext.Status.ResolvedBundle = bundleMetadataFor(bundle)
 	setResolvedStatusConditionSuccess(&ext.Status.Conditions, fmt.Sprintf("resolved to %q", bundle.Image), ext.GetGeneration())
 
+	// Populate the deprecation status using the resolved bundle
+	SetDeprecationStatusInExtension(ext, bundle)
+
```

Comment:
Based on the current state of the logic in determining deprecation status, and without increasing scope to include modifications to it, I think this shouldn't be called until installation has succeeded.

There is some nuance where we theoretically could set the package and/or channel deprecation status after resolution but IMO any changes to the existing deprecation logic is out of scope of this PR.